Style revisions

LiamPattinson opened the pull request:
I used black to reformat the code, and added some manual fixes. I then used flake8 to hunt down any unused/undefined variables. The intention here was to make the code easier to read to assist with future updates (plus black formatting reduces the burden on the programmer to keep the code looking tidy). I tried to ensure that no behaviour was changed in this update, but there are a few changes:

- Number of points in the s-alpha plot reduced to 10x10 to ensure testing can be completed in a reasonable timescale.
- Some parts where the code drops the user into a debugger replaced by exceptions/warnings.
- A few formatted strings converted to f-strings to help reduce line length.

Some issues:

- There are undefined variables in use, and I don't know how to fix the issue:
    - `lmns`, line 199 in profile_plotter
    - `R_ball`/`Z_ball`, lines 284 and 286 in bishoper_save_GX
- There were many unused variables in bishoper_ball. These may have been present for debugging purposes, in which case it may be a good idea to convert the script to a function, add `debug = False` as an argument, and reintroduce them along with some sort of logging/terminal output.

Since the diffs for each file are dominated by black reformatting, I'd recommend checking each of the later commits individually to ensure I haven't changed something important. Some of the longer one-liners have been reformatted into something quite ugly, but I'm intending to refactor those lines to improve readability in future updates.

Let me know if you're unhappy with any of this. If the default black line length is too short for you, I could go up to around 120 characters, but if it's any longer than that I start to find things difficult to read. I usually stick with the default 88 characters since it allows me to comfortably fit three files in my editor at once, but I'm aware that other programmers find it a bit restrictive.

LiamPattinson committed:
Black reformatting

Black formatting, plus manual work to get comments/strings within
88 chars. Removed a small amount of commented-out code where it
couldn't be easily fit (looked like old debugging lines). Changed
some formatted strings to f-strings.

Black has done some strange things to long one liners. A later
more detailed review will hopefully fix those issues.

LiamPattinson committed:
Fixed utils.py flake8 warnings, removed debugging bits

Changed function comments to docstrings, added some TODO
comments for future refactors. Sections where the code dropped
the user into a debugger now raise an exception/warning.

LiamPattinson committed:
Fixed eikcoefs_final.py flake8 errors

LiamPattinson committed:
Fixed profile_plotter.py flake8 issues, bug remains

Line 199, lmns is not defined.

LiamPattinson committed:
Removed unused variables from bishoper_ball.py

LiamPattinson committed:
Flake8 fixes for bishoper_save_GS2/GX

Some undefined variables found in bishoper_save_GS2/GX. Some
were clearly typos and have been corrected, but some remain
in bishoper_save_GX.

rahulgaur104 commented:
This looks good. I'll go throught the individual commits later. Thanks, Liam! 
Yes, lmns is not needed in profile plotter.py. The purpose of lmns is to convert the theta used by VMEC into a straight-field-line theta coordinate. 
Yes, R_ball and Z_ball were put there for the purpose of debugging that function. Moving all the unused variable under a debug flag is a great idea.

LiamPattinson committed:
Cleaned up profile_plotter, implemented parse_input_file

Removed a lot of unused variables from profile_plotter (flake8 didn't
catch them first time around). Reorganised the file so that data
is extracted from the netcdf closer to where it's used. Converted
to a runnable module using the 'if __name__ == "__main__"' idiom.

LiamPattinson committed:
Added default output_dir to plot_profiles

LiamPattinson commented:
I've made some more detailed edits to `profile_plotter.py`, reducing it to the minimum needed to produce the plots. A lot of the data extraction to get `R`, `Z`, `P` etc from the VMEC netcdf looks similar to that in `eikcoefs_final.py`, so longer term I think it'd be good to write a single function call to be used in both files that extracts everything into a dict (or maybe an [xarray](https://docs.xarray.dev/en/stable/) dataset would be better?)

I've also used the `if __name__ == "__main__"` idiom to turn it into a runnable module. It's now possible to call `plot_profiles` from other Python files by giving just the name of a VMEC file.

I also added a function to `utils.py` that reads `eikcoefs_final_input.txt` using some built-in tools. The format you've been using was an exact match for the files used by [configparser](https://docs.python.org/3/library/configparser.html), except that it needed a header at the top of the file. I switched the use of `eval` for `ast.literal_eval`, which gives some protection against arbitrary code execution but still lets you read strings, lists, tuples, etc. Otherwise it should behave the same, but with the safety that it'll throw an error if something is really wrong with the input file.

I may leave this PR here, and switch to another for any further updates, as my next tasks will be implementing the `if __name__ == "__main__"` idiom for all other scripts, setting up a packaging system, and starting to write some unit tests. I'll be jumping off from here, but let me know if you'd like any further changes regarding formatting (I can always rebase later). I'll raise a new issue to reintroduce debugging bits later, as it'll be easier to do that after some further development.
